Add manifest for Flatpak build

jwakely opened the pull request:
As described in the README change, this allows you to build a bundled Flatpak application. On Fedora, that would be done like this:

    sudo dnf install flatpak-builder
    flatpak-builder --user --install flatpak/build flatpak/io.github.dagargo.Elektroid.yaml

Once installed, you can run it like:

    flatpak run io.github.dagargo.Elektroid

or for the CLI:

    flatpak run io.github.dagargo.Elektroid --cli info

or equivalently:

    flatpak run --command=elektroid-cli io.github.dagargo.Elektroid info

If this gets added to the repo, I think you would be able to [submit Elektroid to Flathub](https://github.com/flathub/flathub/wiki/App-Submission#how-to-submit-an-app) so that users can install it simply by running `flatpak install io.github.dagargo.Elektroid` (no need to download/clone anything and build from source). That might make it accessible to more users than the https://copr.fedorainfracloud.org/coprs/jwakely/elektroid/ packages I provide for Fedora and CentOS, but I've only lightly tested the flatpak on non-Fedora systems so far.

jwakely commented:
The name of a flatpak application should be a unique ID based on a reversed DNS name, so I used `io.github.dagargo.Elektroid` since `https://dagargo.github.io/elektroid` would be the website for this repo (if you added HTML files to a branch called `gh-pages`). If you have your own domain that you'd prefer to use for that ID then using that would make sense too.

I put the .yaml file in a `flatpak` sub-directory but there's no reason it couldn't be at the top level (or anywhere else) if you prefer.

jwakely commented:
I think this should also install a .desktop file so that the flatpak can be launched from the system menu. I'll change the pull request to "draft" while I figure out how to do that ...

jwakely commented:
I've created a test flatpak repo to demo this:

```
flatpak --user remote-add --no-gpg-verify kayari https://kayari.org/flatpaks/
flatpak --user install Elektroid
```

This will install the flatpak I built using the new manifest file in the pull request, but it's basically installing untrusted software. Only install it if you trust me!

dagargo commented:
This is super cool.

I'll take a proper look tomorrow as I think it needs some attention.

jwakely commented:
Take your time - I did this to solve a problem I had, and that's solved now. This could wait until after the 2.5 release if you don't have time for it now.

dagargo commented:
Sorry for taking longer than anticipated.

I've decided to go with your suggestion and publish Elektroid on https://dagargo.github.io/elektroid. I'm still working on it. But I'll guess tomorrow I'll review this draft.

dagargo commented:
> I think you would be able to [submit Elektroid to Flathub](https://github.com/flathub/flathub/wiki/App-Submission#how-to-submit-an-app)

Once this is merged, I'll absolutely do it. Flatpack would help some people to run Elektroid so publishing it is a must.

> This could wait until after the 2.5 release if you don't have time for it now.

I was just tying the loose ends. We can merge it now and update the required tags for version and releases once the release is out.

BTW, thanks for taking the time to add Flatpack support. :blue_heart:

jwakely commented:
Awesome, thanks for merging it!

Let me know if I can help with building/testing the flatpak or with flathub. I think this would be very helpful for users, but I don't want to cause lots more work for you. You're already doing all the dev work of this great tool.

dagargo commented:
> thanks for merging it!

My pleasure.

> I don't want to cause lots more work for you.

Just wanted to let you know this is not the case.

dagargo commented:
Oops...

I added the file `flatpack/add-releases-to-appdata.patch` but it's not really needed, right?

BTW, I added you to the THANKS file.

jwakely commented:
> I added the file `flatpack/add-releases-to-appdata.patch` but it's not really needed, right?

Right, because you already merged #95 - I've just submitted #96 to fix that.

> BTW, I added you to the THANKS file.

Cool, thanks :)

By the way, I see these warnings when building the flatpak:

```
Finishing app
Exporting share/applications/io.github.dagargo.Elektroid.desktop
Not exporting share/icons/hicolor/scalable/apps/elektroid.svg, non-allowed export filename
Not exporting share/icons/hicolor/scalable/apps/elektroid-symbolic.svg, non-allowed export filename
Not exporting share/icons/hicolor/scalable/apps/elektroid-wave-symbolic.svg, non-allowed export filename
Not exporting share/icons/hicolor/scalable/apps/elektroid-sequence-symbolic.svg, non-allowed export filename
Not exporting share/icons/hicolor/scalable/apps/elektroid-project-symbolic.svg, non-allowed export filename
Not exporting share/icons/hicolor/scalable/apps/elektroid-sound-symbolic.svg, non-allowed export filename
Exporting share/icons/hicolor/scalable/apps/io.github.dagargo.Elektroid.svg
Exporting share/icons/hicolor/scalable/apps/io.github.dagargo.Elektroid-symbolic.svg
Not exporting share/icons/hicolor/icon-theme.cache, wrong extension
Exporting share/appdata/io.github.dagargo.Elektroid.appdata.xml
Please review the exported files and the metadata
```

I'll review the "not exporting" lines and see what that means and how to fix it.
